src/format/multifactor/AuthenticationFactor.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2024 KeePassXC Team <team@keepassxc.org>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 or (at your option)
+ * version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AuthenticationFactor.h"
+#include "AESCBCFactorKeyDerivation.h"
+
+#include <QDebug>
+#include <QSharedPointer>
+
+void AuthenticationFactor::setName(const QString& name)
+{
+    m_name = name;
+}
+
+const QString& AuthenticationFactor::getName() const
+{
+    return m_name;
+}
+
+void AuthenticationFactor::setKeyType(AuthenticationFactorKeyType type)
+{
+    m_keyType = type;
+
+    if (m_keyType == AuthenticationFactorKeyType::AES_CBC) {
+        m_derivation = QSharedPointer<AESCBCFactorKeyDerivation>::create();
+    } else {
+        m_derivation = QSharedPointer<AESCBCFactorKeyDerivation>(nullptr);
```

Comment:
This is somewhat poor behavior from an API standpoint. We should be returning true/false if the key type is set correctly so callers can know that they just invalidated the class contract. Also simply called `m_derivation.reset()` is a cleaner way to set it to nullptr.